src/lib.rs

```diff
@@ -339,6 +339,27 @@ pub struct Buffer {
     //detail: <C as Context>::BufferDetail,
 }
 
+/// A description of what portion of a buffer to use
+pub struct BufferSlice<'a> {
+    buffer: &'a Buffer,
+    start_bound: Option<BufferAddress>,
```

Comment:
why don't we call them `offset: BufferAddress` and `size: Option<BufferAddress>`? No need to have `start_bound` as optional.